rockcraft/models/project.py

```diff
@@ -326,6 +327,19 @@ def _validate_base_and_overlay(cls, item: Dict[str, Any], values) -> Dict[str, A
             )
         return item
 
+    @pydantic.validator("entrypoint_service", always=True)
+    @classmethod
+    def _validate_entrypoint_service(
+        cls, entrypoint_service: Optional[str], values: Any
+    ) -> Optional[str]:
+        """Verify that the entrypoint_service exist in the services dict."""
+        services = values.get("services")
+        if entrypoint_service and not (
+            services and entrypoint_service in services.keys()
+        ):
+            raise ProjectValidationError("entrypoint_service used is invalid!")
```

Comment:
The user will see this message so it's better to be a bit more verbose (e.g. "The provided entrypoint-service is not a valid Pebble service"). Also, it should be `entrypoint-service` instead of `entrypoint_service` cause that's the internal variable name, and the users are not exposed to that